sympy/matrices/matrices.py

```diff
@@ -26,6 +26,8 @@
 
 def _iszero(x):
     """Returns True if x is zero."""
+    if len(x.free_symbols) == 0:
```

Comment:
It should not be necessary to simplify everything that has no free symbols. In many cases ``x.is_zero`` gives a result without simplification. For example,
```
>>> x = Integer(2)
>>> x.is_zero
False
```